File: frontend/smart-home-ui/src/app/historymodels/commandhistoryresponse.ts
/** API response shape for a single command history entry. */
export interface CommandHistoryResponse {
  /** Unique history entry identifier (UUID). */
  id: string;
  /** Name of the command that was executed. */
  commandName: string;
  /** ISO timestamp of when the command was executed. */
  timestamp: string;
}

File: backend/src/SmartHome.Api/Devices/CommandHistoryResponse.cs
namespace SmartHome.Api.Devices;

/// <summary>
/// DTO used to represent a command history entry returned by the API.
/// </summary>
public class CommandHistoryResponse
{
    /// <summary>Identifier of the device this history entry belongs to.</summary>
    public Guid DeviceId { get; set; }
    /// <summary>Human-readable description of the command that was executed.</summary>
    public string CommandName { get; set; } = string.Empty;
    /// <summary>UTC timestamp of when the command was executed.</summary>
    public DateTime Timestamp { get; set; }
}
